source/D2Lang/src/D2Unicode/D2UnicodeStr.cpp

```diff
@@ -85,3 +85,47 @@ int __fastcall Unicode::strlen(const Unicode* str) {
 
   return i;
 }
+
+Unicode* __fastcall Unicode::strstr(
+    const Unicode* str,
+    const Unicode* substr) {
+  if (str[0].ch == L'\0' || substr[0].ch == L'\0') {
+    return NULL;
+  }
+
+  size_t i_str;
+  for (i_str = 0; ; ++i_str) {
+    for (; str[i_str].ch != substr[0].ch; ++i_str) {
+      if (str[i_str].ch == L'\0') {
+        return NULL;
+      }
+    }
+
+    if (str[i_str].ch == '\0') {
+      return NULL;
+    }
+
+    size_t i_substr;
+    for (i_substr = 0;
```

Comment:
Could we use `Unicode::strcmp` here instead to simplify the code ?

Reply:
`Unicode::strcmp` won't work in this case, since `Unicode::strcmp` needs the null-terminating character of the string and the substring to be at the same relative position. It may be more appropriate to use `Unicode::strncmp`, but would require adding that function first, and calling `Unicode::strlen` to get the length of the substring.

Reply:
As an addendum to my prior comment,
```C
`int __fastcall Unicode::strncmp(const Unicode* str1, const Unicode* str2, size_t count);`
```
suffers from a vanilla bug. If `count` is greater than the length of either string, and one string is a substring of the other, then the strings are determined to be the same.